### PR TITLE
[TensorExpr] Nuke TE shape inference.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65555
* #65859
* #65852
* #65843
* __->__ #65554

We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -384,209 +384,12 @@ std::vector<ExprHandle> TensorExprKernel::sizesForValue(
...
-    const std::vector<ExprHandle>& b) {
-  auto res = broadcastShapesImpl(a, b);
-  if (res.second) {
-    hasBroadcast_ = true;
-  }
-  return res.first;
+  GRAPH_DEBUG("Unknown sizes for the node: ", *v->node());
+  GRAPH_DEBUG("Full fusion group graph:\n", *v->node()->owningGraph());
+  std::string msg =
+      std::string("Unhandled node kind (in inferSizesForValue): ") +
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/kernel.cpp:390]: Update this function name.

**Review by navahgar — COMMENTED** [review]
Do we need more tests for both positive and negative cases here? Just to make sure that we don't throw exceptions in some corner cases.

**Comment by bertmaher** [comment]
Broadcast + random almost certainly does not work correctly.  I think we do not fuse rand for this reason

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
> Do we need more tests for both positive and negative cases here? Just to make sure that we don't throw exceptions in some corner cases.

Do you mean tests where shapes are not specified in the graph? I think this situation doesn't occur in the current setting (and that's the reason I'd like to kill this code - it's not actually used). In the near future I plan to add some passes that would verify that graph has all the info so that we can be sure TEK can handle it. Would it be ok with you if I add more tests along with that pass?

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by bertmaher** [comment]
I'm slightly frightened of this change since it reduces the number of if statements in front of a `throw` :-D (I kind of suspect @navahgar feels the same).  I'm wondering if we should backtest it on a bunch of production models...

**Comment by ZolotukhinM** [comment]
I'm down to testing it just in case, I'll follow up with you guys to learn how exactly to do that :)

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Review by bertmaher — APPROVED** [review]

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

**Pull request closed without merging**